src/main/java/seedu/duke/operationlist/PassengerList.java

```diff
@@ -0,0 +1,453 @@
+    public void addOperation(String passengerDetail) throws SkyControlException {
+        getNumberOfPassengers();
+        getPassengerDetails(passengerDetail);
+        validateDetailFormat();
+        checkPassengerDetails();
+        checkPassengerDuplicate();
+        PassengerInfo passenger = new PassengerInfo(name, departureTime, flightNumber,
+                gateNumber, boardingGroup, seatNumber, boardingTime);
+        passengers.add(passenger);
+        Ui.showAddedPassenger(passenger);
+    }
+
+    @Override
+    public void delayFlightDeparture(String flightNum, String newDepartureTime) {
+        modifyPassengersDepartureTime(flightNum, newDepartureTime);
+    }
+
+    @Override
+    public void modifyFlightNum(String flightNum, String newFlightNum) {
+        modifyPassengersFlightNum(flightNum, newFlightNum);
+    }
+
+    @Override
+    public void modifyGateNum(String flightNum, String newGateNum) {
+        modifyPassengersGateNum(flightNum, newGateNum);
+    }
+
+    //@@author ivanthengwr
+    private void checkPassengerDetails() throws SkyControlException {
+        if (isExceedNameLength) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getExceedNameLengthError(name));
+        } else if (isWrongNameFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getNameError());
+        } else if (isWrongDepartureFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getDepartureTimeError());
+        } else if (isWrongFlightNumFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getFlightNumberError());
+        } else if (isWrongBoardingTimeFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getBoardingTimeError());
+        } else if (isWrongGateNumberFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getGateNumberError());
+        } else if (isWrongBoardingGroupFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getBoardingGroupError());
+        } else if (isWrongSeatNumberFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getSeatNumberError());
+        }
+    }
+
+    //@@author shengiv
+    private void checkPassengerDuplicate() throws SkyControlException {
+        for (int i = 0; i < numOfPassengers; i++) {
+            validatePassenger(i);
+            if (isPassengerDuplicate()) {
+                resetCheckDuplicate();
+                throw new SkyControlException(ui.getDuplicatePassengerError());
+            }
+        }
+    }
+
+    private void modifyPassengersFlightNum(String flightNum, String newFlightNum) {
+        getNumberOfPassengers();
+        for (int i = 0; i < numOfPassengers; i++) {
+            PassengerInfo passenger = passengers.get(i);
+            if (passenger.getFlightNumber().equals(flightNum)) {
```

Comment:
Avoid complicated expressions. Also to minimise the scope, consider putting value of `passenger.getFlightNumber()` into a boolean variable right before using it to check for the flightNum. The same applies to the other boolean checks.